src/components/PrimaryNav/PrimaryNav.tsx

```diff
@@ -1,82 +1,79 @@
 import { dashboardUpdateAvailable } from "@canonical/jujulib/dist/api/versions";
 import { Icon, StatusLabel, Tooltip } from "@canonical/react-components";
-import classNames from "classnames";
 import { useEffect, useMemo, useRef, useState } from "react";
 import { useSelector } from "react-redux";
-import { NavLink } from "react-router-dom";
 
 import Logo from "components/Logo/Logo";
 import UserMenu from "components/UserMenu/UserMenu";
-import { getAppVersion } from "store/general/selectors";
+import { getAppVersion, getIsJuju } from "store/general/selectors";
 import {
   getControllerData,
   getGroupedModelStatusCounts,
 } from "store/juju/selectors";
 import type { Controllers } from "store/juju/types";
 import urls, { externalURLs } from "urls";
+
 import "./_primary-nav.scss";
+import PrimaryNavLink from "./PrimaryNavLink";
 
 const ModelsLink = () => {
   const { blocked: blockedModels } = useSelector(getGroupedModelStatusCounts);
   return (
-    <NavLink
-      className={({ isActive }) =>
-        classNames("p-list__link", {
-          "is-selected": isActive,
-        })
-      }
-      to={urls.models.index}
-    >
-      <i className={`p-icon--models is-light`}></i>
-      <span className="hide-collapsed">Models</span>
+    <PrimaryNavLink to={urls.models.index} iconName="models" title="Models">
       {blockedModels > 0 && (
         <span className="entity-count is-negative">{blockedModels}</span>
       )}
-    </NavLink>
+    </PrimaryNavLink>
   );
 };
 
 const ControllersLink = () => {
   const controllers: Controllers | null = useSelector(getControllerData);
 
   const controllersUpdateCount = useMemo(() => {
-    if (!controllers) return 0;
-    let count = 0;
-    Object.values(controllers).forEach((controller) => {
-      controller.forEach((controller) => {
-        if ("version" in controller && controller.updateAvailable) {
-          count += 1;
-        }
-      });
-    });
-    return count;
+    if (!controllers) {
+      return 0;
+    }
+    return Object.values(controllers).reduce(
+      (controllersCount, controllerArray) =>
+        controllersCount +
+        controllerArray.reduce(
+          (count, controller) =>
+            "version" in controller && controller.updateAvailable
+              ? count + 1
+              : count,
+          0
+        ),
+      0
+    );
```

Comment:
Same functionality, but refactored code. Let me know if previous version is better.